prov/efa/src/rdm/rxr_ep.h

```diff
@@ -279,6 +285,57 @@ static inline int rxr_need_sas_ordering(struct rxr_ep *ep)
 	return ep->msg_order & FI_ORDER_SAS;
 }
 
+static inline uint64_t rxr_get_host_id(char *host_id_file)
+{
+	FILE *fp = NULL;
+	char *host_id_str = NULL;
+	const size_t HOST_ID_LENGTH = 16;
+	const size_t HOST_ID_PREFIX_LENGTH = 3; /* Ignore the i-0 sequence */
+	size_t length = 0;
+	uint64_t host_id = 0;
+
+	if (!host_id_file) {
+		EFA_WARN(FI_LOG_EP_CTRL, "Host id file is not specified\n");
+		goto out;
+	}
+
+	fp = fopen(host_id_file, "r");
+	if (!fp) {
+		EFA_WARN(FI_LOG_EP_CTRL, "Cannot open host id file: %s\n", host_id_file);
+		goto out;
+	}
+
+	fseek(fp, HOST_ID_PREFIX_LENGTH, SEEK_SET);
+
+	host_id_str = calloc(HOST_ID_LENGTH, sizeof(*host_id_str));
+	if (!host_id_str) {
+		EFA_WARN(FI_LOG_EP_CTRL, "Cannot allocate memory\n");
+		goto out;
+	}
+
+	length = fread(host_id_str, 1, HOST_ID_LENGTH, fp);
+	if (length != HOST_ID_LENGTH) {
+		EFA_WARN(FI_LOG_EP_CTRL, "Failed to read host id. Read length: %lu Expect length: %lu\n", length, HOST_ID_LENGTH);
+		goto out;
+	}
+
+	for (char *s = host_id_str; *s; ++s) {
```

Comment:
No need to do validation by yourself. `strtoul` can do it for you. Just pass a pointer to `strtoul` instead of NULL. and make sure the pointer is pointing to the end of the string

Reply:
Thanks! Addressed.